dune/harmonizer/dunesql/dunesql.py

```diff
@@ -42,6 +43,7 @@ class Generator(Trino.Generator):
                     cast_boolean_strings,
                     cast_date_strings,
                     replace_0x_strings_with_hex_strings,
+                    remove_varchar_lengths_in_casts,
```

Comment:
Hey 👋

I don't think this is correct because it'll also cause the `VARCHAR` length to be dropped when translating Postgres queries to Dune SQL, for example. However, Postgres semantics are different from Spark's in this case:

```
georgesittas=# select cast(123456 as varchar(3));
 varchar
---------
 123
(1 row)
```

I believe the correct approach to this problem is to drop the length at parse time just for the Spark dialect, so that the produced AST reflects the correct semantics. This will automatically solve the transpilation issue, too.

On that note, maybe this is something we can fix upstream since Spark seems to allow `CHAR` and `VARCHAR` types only for schema definitions; for queries it uses the `STRING` type, as far as I understand from their [docs](https://spark.apache.org/docs/latest/sql-ref-datatypes.html):

<img width="803" alt="Screenshot 2023-06-05 at 4 11 41 PM" src="https://github.com/duneanalytics/harmonizer/assets/46752250/2da83c4c-f9e1-4814-9e25-01a739724e1c">

cc: @vegarsti